runtime/src/threads/pthreads/threads-pthreads.c

```diff
@@ -81,6 +83,102 @@ static void            (*saved_threadEndFn)(void);
 static void*           initial_pthread_func(void*);
 static void*           pthread_func(void*);
 
+/* This conditional is necessary because on some operating systems
+ * (like OSX) the cstdlib malloc doesn't use nmap for the allocation
+ * and can't be mprotect-ed. In that case, we use the default pthread
+ * stack
+ */
+#ifndef CHPL_USING_CSTDLIB_MALLOC
```

Comment:
I'd rather have registered stacks without guard pages when the heap is on huge pages. But, for Mac OS X, I think I'd rather have the system-provided guard pages (since there is a problem getting pages that we can mprotect). Alternatively, on that system only, we could allocate with mmap...

I don't understand how the question about huge pages impacts this code. AFAIK we only use huge pages in order to create our registered heap; so a huge page configuration would have allocator != cstdlib. So I don't see what's actually wrong with the functionality from this ifdef approach, other than that it could be prettier...

Maybe the change that we want is just to disable the guard page creation if we're on huge pages? And leave the other logic the way it is (although consider writing it differently?)